src/aap_eda/api/serializers/credential.py

```diff
@@ -37,9 +37,20 @@ class Meta:
         ]
 
 
-class CredentialCreateSerializer(serializers.ModelSerializer):
-    secret = serializers.CharField(required=True, allow_null=False)
+def _validate_scm_ssh_key_params(ssh_key, ssh_key_passphrase):
+    """Both, or neither, of ssh key-related values must be specified."""
+    if ((ssh_key is not None) and (ssh_key != "")) and (
```

Comment:
passphrase is optional. You should be able to use a private key without passphrase.